litert/vendors/intel_openvino/dispatch/device_context.cc

```diff
@@ -0,0 +1,187 @@
+  LITERT_RETURN_IF_ERROR(
+      !tensor_type.layout.has_strides,
+      litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                         "Tensor strides are not supported"));
+
+  switch (tensor_buffer_type) {
+    case kLiteRtTensorBufferTypeDmaBuf: {
+#if LITERT_HAS_DMABUF_SUPPORT
+      int buffer_fd;
+      void *buffer_host_addr;
+      LITERT_RETURN_IF_ERROR(
+          LiteRtGetTensorBufferDmaBufBuffer(tensor_buffer, &buffer_host_addr,
+                                            &buffer_fd),
+          litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                             "Failed to get DMA-BUF buffer"));
+
+      auto mmap_handle = mmap(NULL, tensor_buffer_size, PROT_WRITE | PROT_READ,
+                              MAP_SHARED, buffer_fd, tensor_buffer_offset);
+
+      if (mmap_handle == MAP_FAILED)
+        return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                  "MMAP failed for tensor buffer");
+
+      auto context = core_.get_default_context("NPU")
+                         .as<ov::intel_npu::level_zero::ZeroContext>();
+      std::vector<int32_t> ov_shape_vec(tensor_type.layout.rank);
+      for (int i = 0; i < ov_shape_vec.size(); i++)
+        ov_shape_vec[i] = tensor_type.layout.dimensions[i];
+
+      // TODO: change f32 to ov_element_type fetched from TensorType
+      auto remote_tensor = context.create_tensor(
+          ov::element::f32, ov::Shape{ov_shape_vec.begin(), ov_shape_vec.end()},
+          buffer_fd);
+      tensor_handle_map_.emplace((LiteRtTensorBufferHandle)mmap_handle,
+                                 remote_tensor);
+      return (uint64_t)mmap_handle;
+
+#else
+      return litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                "DmaBuf support is missing on this platform");
+#endif  // LRT_HAS_DMABUF_SUPPORT
+      break;
+    }
+    case kLiteRtTensorBufferTypeAhwb: {
+#if LITERT_HAS_AHWB_SUPPORT
+      AHardwareBuffer *ahwb;
+      LITERT_RETURN_IF_ERROR(
+          LiteRtGetTensorBufferAhwb(tensor_buffer, &ahwb),
+          litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                             "Failed to get LiteRT Tensor Buffer for AHWB"));
+
+      auto fd_exp = GetFdFromUnixHandle(ahwb);
+      int fd = fd_exp.Value();
+      LITERT_RETURN_IF_ERROR(
+          fd != -1, litert::Unexpected(kLiteRtStatusErrorRuntimeFailure,
+                                       "Failed to get FD from unix handle"));
+
+      std::vector<int32_t> ov_shape_vec(tensor_type.layout.rank);
+      for (int i = 0; i < ov_shape_vec.size(); i++)
+        ov_shape_vec[i] = tensor_type.layout.dimensions[i];
+      auto context = core_.get_default_context("NPU")
+                         .as<ov::intel_npu::level_zero::ZeroContext>();
+      ov::RemoteTensor tensor = context.create_tensor(
+          ov::element::f32, ov::Shape{ov_shape_vec.begin(), ov_shape_vec.end()},
+          fd);
+      static int mmap_handle = 1;
```

Comment:
Can you guarantee that the handle is not overwrapped with LITERT_HAS_DMABUF_SUPPORT ?

How about using `std::vector<ov::RemoteTensor>` and use the index as handle?

```
int handle = tensor_handle_vector_.size();
tensor_handle_vector_.push_back(tensor);
return handle;
```

Reply:
I can still see the map usage.

Reply:
I find it better to manage the map creation and destruction for storing the RemoteTensor objects rather than a vector. It is a cleaner approach while destroying the object individually.